Packs/CommonScripts/Scripts/SetWithTemplate/SetWithTemplate.py

```diff
@@ -0,0 +1,259 @@
+        if not ok:
+            for sep in ['=', '(', '.']:
+                name = key.split(sep, maxsplit=1)[0]
+                ok, val = execute_command('getList', {'listName': name}, fail_on_error=False)
+                if ok:
+                    break
+            else:
+                return None
+
+        val = demisto.dt(val, '.' + key[len(name):])
+        if isinstance(val, str):
+            try:
+                val = json.loads(val)
+            except json.JSONDecodeError:
+                pass
+        return val
+
+    def get(self, key: Optional[str] = None) -> Any:
+        """ Get the context value
+
+        :param key: The dt expressions (string within ${}).
+        :return: The value.
+        """
+        if not key:
+            return None
+
+        for prefix, handler in {
+            'inputs': ContextData.__get_inputs_value,
+            'incident': ContextData.__get_incident_value,
+            'lists': ContextData.__get_lists_value
+        }.items():
+            if prefix == key or (key.startswith(prefix)
+                                 and key[len(prefix):len(prefix) + 1] in ('.', '(', '=')):
+                return handler(self, key[len(prefix):])
+        return demisto.dt(self.__context, key)
+
+
+class Formatter:
+    def __init__(self, start_marker: str, end_marker: str, keep_symbol_to_null: bool):
+        if not start_marker:
+            raise ValueError('start-marker is required.')
+
+        self.__start_marker = start_marker
+        self.__end_marker = end_marker
+        self.__keep_symbol_to_null = keep_symbol_to_null
+
+    @staticmethod
+    def __is_end_mark(source: str, ci: int, end_marker: str) -> bool:
+        if end_marker:
+            return source[ci:ci + len(end_marker)] == end_marker
+        else:
+            c = source[ci]
+            if c.isspace():
+                return True
+            elif c.isascii():
+                return c != '_' and not c.isalnum()
+            else:
+                return False
+
+    def __extract(self,
+                  source: str,
+                  extractor: Optional[Callable[[str,
+                                                Optional[ContextData]],
+                                               Any]],
+                  dx: Optional[ContextData],
+                  si: int,
+                  end_marker: Optional[str]) -> Tuple[Any, Optional[int]]:
+        out = None
+        ci = si
+        while ci < len(source):
+            if end_marker is not None and Formatter.__is_end_mark(source, ci, end_marker):
+                key = source[si:ci] if out is None else str(out) + source[si:ci]
+                xval = ''
+                if extractor:
+                    xval = extractor(key, dx)
+                    if self.__keep_symbol_to_null and xval is None:
+                        xval = self.__start_marker + key + self.__end_marker
+                return xval, ci + len(end_marker)
```

Comment:
Should there be a return here?
if I'm not mistaken:
```python
__extract('> Hello my friend', extract_dt, {}, 0, '>')
```
will result in `ci=0`, `key='>'`, and `xval=None`
which will return `None, 1`

Reply:
@yaakovi That's expected behavior. `end_marker` is only given for parsing the value within ${}. So the extracted value immidiately returns when the end marker is found.

Reply:
I've added a docstring for it.